spec/system/task_spec.rb

```diff
@@ -1,24 +1,25 @@
 require 'rails_helper'
 
 RSpec.describe 'Task', type: :system do
+  let(:project) { FactoryBot.create(:project) }
+  let(:task) { FactoryBot.create(:task, project_id: project.id) }
```

Comment:
Factoryにassociationを定義することで、project_idの指定を省略できます。
```suggestion
  let(:task) { create(:task) }
```